driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/gridfs/GridFSUploadPublisherImpl.java

```diff
@@ -148,47 +145,50 @@ public void subscribe(final Subscriber<? super ObjectId> subscriber) {
         };
     }
 
-    private Mono<Void> createCheckAndCreateIndexesMono() {
-        MongoCollection<Document> collection = filesCollection.withDocumentClass(Document.class).withReadPreference(primary());
-        FindPublisher<Document> findPublisher;
-        if (clientSession != null) {
-            findPublisher = collection.find(clientSession);
-        } else {
-            findPublisher = collection.find();
-        }
+    private Mono<Void> createCheckAndCreateIndexesMono(@Nullable final Timeout timeout) {
         AtomicBoolean collectionExists = new AtomicBoolean(false);
-
-        return Mono.create(sink -> Mono.from(findPublisher.projection(PROJECTION).first())
-                .subscribe(
-                        d -> collectionExists.set(true),
+        return Mono.create(sink -> findAllInCollection(filesCollection, timeout).subscribe(
+                        d -> {
+                            collectionExists.set(true);
+                            },
                         sink::error,
                         () -> {
                             if (collectionExists.get()) {
                                 sink.success();
                             } else {
-                                checkAndCreateIndex(filesCollection.withReadPreference(primary()), FILES_INDEX)
-                                        .doOnError(sink::error)
+                                checkAndCreateIndex(filesCollection.withReadPreference(primary()), FILES_INDEX, timeout)
                                         .doOnSuccess(i -> {
-                                            checkAndCreateIndex(chunksCollection.withReadPreference(primary()), CHUNKS_INDEX)
-                                                    .doOnError(sink::error)
-                                                    .doOnSuccess(sink::success)
-                                                    .subscribe();
+                                            checkAndCreateIndex(chunksCollection.withReadPreference(primary()), CHUNKS_INDEX, timeout)
```

Comment:
I've updated the code from .`doOnError(sink::error).doOnSuccess(sink::success).subscribe();` to `.subscribe(unused -> {}, sink::error, sink::success);`. This change is intended to ensure that error signals are properly propagated.